src/textual/reactive.py

```diff
@@ -169,53 +159,41 @@ def _reset_object(cls, obj: object) -> None:
         getattr(obj, "__watchers", {}).clear()
         getattr(obj, "__computes", []).clear()
 
-    def __set_name__(self, owner: Type[MessageTarget], name: str) -> None:
-
+    def __set_name__(self, owner: Reactable, name: str) -> None:
         # Check for compute method
         if hasattr(owner, f"compute_{name}"):
             # Compute methods are stored in a list called `__computes`
-            try:
-                computes = getattr(owner, "__computes")
-            except AttributeError:
+            if not hasattr(owner, "__computes"):
```

Comment:
faster than raising an exception. Not a big deal tbh, since this happens once, but still more dogmatic. (

Reply:
Probably the idea here is that `__computes` is an attribute that is very likely there and `try` statements have near-zero overhead.

Reply:
My gut feeling is that this is slower, because it does two functions calls rather than one for the 90% case. And Rodrigo is correct that newer Pythons (3.10 +) exceptions are negligible overhead.

Reply:
I will revert.